### PR TITLE
5490 shib adfs

### PR DESCRIPTION
**Pull request by pameyer**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #5490 

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+<SPConfig xmlns="urn:mace:shibboleth:3.0:native:sp:config" xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" clockSkew="1800">
+  <!-- adfs libraries for shibboleth and apache -->
+  <OutOfProcess>
+    <Extensions>
+      <Library path="adfs.so" fatal="true"/>
+    </Extensions>
+  </OutOfProcess>
+  <InProcess>
+    <Extensions>
+      <Library path="adfs-lite.so" fatal="true"/>
```
Comment (pdurbin) [inline doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml]: How does one install these .so files? Do they come in the shib RPM?

File: doc/sphinx-guides/source/installation/shibboleth.rst [hunk 2]
```diff
@@ -196,6 +196,12 @@ attribute-map.xml
 
 By default, some attributes ``/etc/shibboleth/attribute-map.xml`` are commented out. Edit the file to enable them so that all the require attributes come through. You can download a :download:`sample attribute-map.xml file <../_static/installation/files/etc/shibboleth/attribute-map.xml>`.
 
+Shibboleth and ADFS
+~~~~~~~~~~~~~~~~~~~
+With appropriate configuration, Dataverse and Shibboleth can make use of "single sign on" using Active Directory.
+This requires configuring ``shibd`` and ``httpd`` to load appropriate libraries, and insuring that the attribute mapping matches those provided.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/shibboleth.rst]: Are the appropriate libraries included in the shib RPM?

File: doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml (hunk 1 above)
Comment (pameyer) [reply doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml]: Probably; will confirm.

Comment (pdurbin) [reply doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml]: @pameyer thanks, but no need. I found this in the docs that says these files do come from the shib RPM:

"The WS-Federation code for the SP is supplied in an extension that is included with the SP source code, and is built by default and included with binary packages. The extension is contained in two libraries, one for shibd and one for the web server filter/module. These are named adfs.so and adfs-lite.so respectively." https://wiki.shibboleth.net/confluence/display/SP3/ADFS

Comment (donsizemore) [reply doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml]: I did nothing special to UNC's Shib to allow ADFS to work with it... just wedged in the custom IDP XML.

**Comment by djbrooke** [comment]
I'm going to move this over to QA, as I was the only remaining reviewer assigned to this.

File: doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml (hunk 1 above)
Comment (pameyer) [reply doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml]: I didn't try with the adfs.so/adfs-lite.so removed; but it's a definite possibility that they aren't needed.

Comment (djbrooke) [reply doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml]: Great work here.

Comment (djbrooke) [reply doc/sphinx-guides/source/_static/installation/files/etc/shibboleth/shibboleth2_adfs.xml]: (was testing something, but great work, for real :))

**Pull request merged**